src/main/scala/symsim/examples/concrete/breaking/Car.scala

```diff
@@ -72,9 +72,7 @@ object Car
     def initialize: Randomized[CarState] = for
       v <- Randomized.repeat (Randomized.between (0.0, 10.0))
       p <- Randomized.repeat (Randomized.between (0.0, 15.0))
-      s0 = CarState (v,p)
-      s <- if isFinal (s0) then initialize
-           else Randomized.const (s0)
+      s = CarState (v, p) if !isFinal (s)
```

Comment:
I checked this change and something is wrong there. Because the learning process is not done correctly.
I attached the q-table after 100K episodes for breaking car.
<img width="968" alt="Screenshot 2022-10-07 at 12 10 17" src="https://user-images.githubusercontent.com/77638469/194530004-b39f95d5-c16e-426a-a038-b99f11073ea0.png">

Reply:
There is no option to add an `else` here.  The `if` in a for-yield (could also be written in a separate line) is just a filter. So for a LazyList it should remove the states violating the conditions.  This seems to work for simplemaze. Can you check? (and also look at the simplemaze's source code, to get an idea what is going on). I will check docs for for-yield in the meantime.

Reply:
I can at least confirm that I see similarly weird training for breaking car on my side.  I will try to write a test whether `initialize` gives a non-constant stream.

Reply:
I checked the simple maze and there are two points in which it works better:
1. The gamma value was 1.0 in the test I found in the simple maze. I decreased it to 0.1 which didn't converge so fast but still was better than breaking car.
2. The state space in the simple maze is discretized (it is continuous for the breaking car), and the probability of selecting a non-final state in initialize function (with the current conditions) is more in comparison with the breaking car. Besides, all of the states in the simple maze, are reachable from others.

Reply:
Finally got to this.  We are clearly not getting good randomness here:

![image](https://user-images.githubusercontent.com/1753628/194777013-5bd1c283-12f9-471d-994d-7c034b6dde86.png)

`s.v` seems constant and `s.p` varies very little

Reply:
I'll push a commit to lower γ for `simplemaze` below 1 (in the same PR). Not sure if `0.1` is not too small, I settled for `0.9`.

Reply:
OK. This was a bit nasty. It turns out I cannot pull from two infinite streams of random numbers, because I get one value from the first one (`v`) and this is paired with infinitely many values of the second one (`p`) and effectively the first value never advances.  Pulling just one value from the second stream (without `repeat`) seems to work, as then we get one `p` value for each `v` value:
![image](https://user-images.githubusercontent.com/1753628/194778060-ffdc9118-a56f-4e05-bea0-9c5d9ce84129.png)
I will push a correction now.  Please test again.

Reply:
Btw. the tables look OK in my terminal, and they are broken in your screenshots.  I start to suspect that there is some Mac incompatibility involved:
![image](https://user-images.githubusercontent.com/1753628/194778200-85fe5086-0833-4ad9-a912-3a1a2333a0e6.png)